buffalo/algo/plsi.py

```diff
@@ -155,54 +156,52 @@ def _iterate(self):
         self.obj.swap()
 
         self.logger.debug(
-            f'updated processed({updated}) elapsed(data feed: {feed_t:0.5f} update: {update_t:0.5f})')
+            f"updated processed({updated}) elapsed(data feed: {feed_t:0.5f} update: {update_t:0.5f})")
         return loss_nume, loss_deno
 
-    def train(self):
+    def train(self, training_callback: Optional[Callable[[int, Dict[str, float]], None]] = None):
         best_loss, loss, self.validation_result = 1e+10, None, {}
-        self.initialize_tensorboard(self.opt.num_iters)
-        self.logger.info(f'Train pLSI, K: {self.opt.d}, alpha1: {self.opt.alpha1}, '
-                         f'alpha2: {self.opt.alpha2}, num_workers: {self.opt.num_workers}')
+        self.logger.info(f"Train pLSI, K: {self.opt.d}, alpha1: {self.opt.alpha1}, "
+                         f"alpha2: {self.opt.alpha2}, num_workers: {self.opt.num_workers}")
         for i in range(self.opt.num_iters):
             start_t = time.time()
             _loss_nume, _loss_deno = self._iterate()
             train_t = time.time() - start_t
 
             loss = _loss_nume / (_loss_deno + self.opt.eps)
-            metrics = {'train_loss': loss}
+            metrics = {"train_loss": loss}
 
             if self.opt.validation and \
                self.opt.evaluation_on_learning and \
                self.periodical(self.opt.evaluation_period, i):
                 start_t = time.time()
                 self.validation_result = self.get_validation_results()
                 vali_t = time.time() - start_t
-                val_str = ' '.join([f'{k}:{v:0.5f}' for k, v in self.validation_result.items()])
-                self.logger.info(f'Validation: {val_str} Elapsed {vali_t:0.3f} secs')
-                metrics.update({'val_%s' % k: v for k, v in self.validation_result.items()})
-
-            self.logger.info('Iteration %d: Loss %.3f Elapsed %.3f secs' % (i + 1, loss, train_t))
-            self.update_tensorboard_data(metrics)
+                val_str = " ".join([f"{k}:{v:0.5f}" for k, v in self.validation_result.items()])
+                self.logger.info(f"Validation: {val_str} Elapsed {vali_t:0.3f} secs")
+                metrics.update({"val_%s" % k: v for k, v in self.validation_result.items()})
+                if training_callback is not None and callable(training_callback):
+                    training_callback(i, metrics)
+            self.logger.info("Iteration %d: Loss %.3f Elapsed %.3f secs" % (i + 1, loss, train_t))
```

Comment:
여긴 반대로 %s 등의 포맷팅을 쓰는데 만약 f-string을 쓴다면 아래처럼 가도 괜찮을것 같습니다.

```suggestion
            self.logger.info(f"Iteration {i + 1}: Loss {loss:.3f} Elapsed {train_t:.3f} secs")
```